src/document/mod.rs

```diff
@@ -160,14 +163,21 @@ impl Document {
                     };
 
                     if !clean {
-                        result.push_str(&format!(
-                            "{} {}{}#{}{}\n",
-                            comment_start, block_start, path, name, comment_end,
-                        ));
+                        if idx == 0 || block.name != blocks[idx - 1].name {
+                            result.push_str(&format!(
```

Comment:
Maybe just determine `let block_sep = if idx == 0 || ...` and then do
```rust
writeln!(&mut result, "{} {}{}#{}{}", comment_start, block_sep, path, name, comment_end).unwrap();
```
here? (`push_str` and `format!` means allocating a temporary string instead of writing directly into `result`.)

Reply:
Is it safe to use `unwrap()` when writing to a string? Just asking because I didn't venture to do it in git-graph/git-igitt.

Reply:
Definitely. The general `fmt::Write` interface needs to propagate errors when writing into files and such, but writing into a growable `String` can only fail due to out of memory and this aborts the process anyway.